assets/js/components/KeyMetrics/ConnectGA4CTAWidget.test.js

```diff
@@ -0,0 +1,215 @@
+			{
+				registry,
+			}
+		);
+		await waitForRegistry();
+		expect( container ).toHaveTextContent( 'NULL' );
+	} );
+
+	it( 'should not render if GA4 is connected', async () => {
+		provideModules( registry, [
+			{
+				slug: 'analytics-4',
+				active: true,
+				connected: true,
+			},
+		] );
+
+		const { container, waitForRegistry } = render(
+			<ConnectGA4CTAWidget Widget={ Widget } WidgetNull={ WidgetNull } />,
+			{
+				registry,
+			}
+		);
+		await waitForRegistry();
+		expect( container ).toHaveTextContent( 'NULL' );
+	} );
+
+	it( 'should not render unless at least 3 analytics dependant metrics are registered', async () => {
+		const keyMetricWidgets = {
+			[ KM_ANALYTICS_LOYAL_VISITORS ]: [ 'analytics-4' ],
+			[ KM_ANALYTICS_NEW_VISITORS ]: [ 'analytics-4' ],
+			[ KM_ANALYTICS_ADSENSE_TOP_EARNING_CONTENT ]: [ 'adsense' ],
+			[ KM_SEARCH_CONSOLE_POPULAR_KEYWORDS ]: [ 'search-console' ],
+		};
+
+		provideKeyMetrics( registry, {
+			widgetSlugs: Object.keys( keyMetricWidgets ),
+		} );
+
+		registry
+			.dispatch( CORE_WIDGETS )
+			.registerWidgetArea( AREA_MAIN_DASHBOARD_KEY_METRICS_PRIMARY, {
+				title: 'Key metrics',
+			} );
+		registry
+			.dispatch( CORE_WIDGETS )
+			.assignWidgetArea(
+				AREA_MAIN_DASHBOARD_KEY_METRICS_PRIMARY,
+				CONTEXT_MAIN_DASHBOARD_KEY_METRICS
+			);
+
+		Object.keys( keyMetricWidgets ).forEach( ( slug ) => {
+			registry.dispatch( CORE_WIDGETS ).registerWidget( slug, {
+				Component: () => <div>Hello test.</div>,
+				modules: keyMetricWidgets[ slug ],
+			} );
+			registry
+				.dispatch( CORE_WIDGETS )
+				.assignWidget( slug, AREA_MAIN_DASHBOARD_KEY_METRICS_PRIMARY );
+		} );
+
+		const { container, waitForRegistry } = render(
+			<ConnectGA4CTAWidget Widget={ Widget } WidgetNull={ WidgetNull } />,
+			{
+				registry,
+			}
+		);
+		await waitForRegistry();
+		expect( container ).toHaveTextContent( 'NULL' );
+	} );
+
+	it( 'should render if widget is not dismissed, user input is completed, GA4 is not connected, and at least 3 analytics dependant metrics are registered', async () => {
+		const keyMetricWidgets = [
+			KM_ANALYTICS_LOYAL_VISITORS,
+			KM_ANALYTICS_NEW_VISITORS,
+			KM_ANALYTICS_TOP_TRAFFIC_SOURCE,
+			KM_ANALYTICS_ENGAGED_TRAFFIC_SOURCE,
+		];
+
+		provideKeyMetrics( registry, {
+			widgetSlugs: keyMetricWidgets,
+		} );
+
+		registry
+			.dispatch( CORE_WIDGETS )
+			.registerWidgetArea( AREA_MAIN_DASHBOARD_KEY_METRICS_PRIMARY, {
+				title: 'Key metrics',
+			} );
+		registry
+			.dispatch( CORE_WIDGETS )
+			.assignWidgetArea(
+				AREA_MAIN_DASHBOARD_KEY_METRICS_PRIMARY,
+				CONTEXT_MAIN_DASHBOARD_KEY_METRICS
+			);
+
+		keyMetricWidgets.forEach( ( slug ) => {
+			registry.dispatch( CORE_WIDGETS ).registerWidget( slug, {
+				Component: () => <div>Hello test.</div>,
+				modules: [ 'analytics-4' ],
+			} );
+			registry
+				.dispatch( CORE_WIDGETS )
+				.assignWidget( slug, AREA_MAIN_DASHBOARD_KEY_METRICS_PRIMARY );
+		} );
```

Comment:
This is duplicated in a few places now. If `registerDefaults` isn't suitable here, perhaps a new utility would make sense? E.g. `provideKeyMetricsWidgetRegistrations`

Reply:
Created `provideWidgetRegistrations`, let me know if that looks good, thank you!